src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -42,5 +42,29 @@ private unsafe string DisplayString
                 }
             }
         }
+
+        public Vector256<byte> AsByte() => Unsafe.As<Vector256<T>, Vector256<byte>>(ref Unsafe.AsRef(in this));
```

Comment:
Perhaps, we can optimize them for dynamic invocations.
```csharp
public Vector256<byte> AsByte() => Avx.IsSupported ? AsByte() : Unsafe.As<Vector256<T>, Vector256<byte>>(ref Unsafe.AsRef(in this));
```

Reply:
I do not think this optimization would improve anything. Also, the performance for dynamic invocations does not really matter.

Reply:
I also think, if that was something we cared about, optimizing `Unsafe.As` to have efficient codegen for blittable types (where `TFrom` and `TTo` are the same size) might be better